sqlglot/transforms.py

```diff
@@ -628,11 +628,30 @@ def eliminate_full_outer_join(expression: exp.Expression) -> exp.Expression:
             expression_copy = expression.copy()
             expression.set("limit", None)
             index, full_outer_join = full_outer_joins[0]
+
+            tables = (
+                full_outer_join.alias_or_name,
+                expression.args["from"].alias_or_name,
+            )
+            join_conditions = full_outer_join.args.get("on") or exp.and_(
+                *[
+                    exp.column(col, tables[0]).eq(exp.column(col, tables[1]))
+                    for col in full_outer_join.args.get("using")
+                ]
+            )
+
             full_outer_join.set("side", "left")
+            anti_join_clause = (
+                exp.Select(expressions=[exp.Literal(this=1, is_string=False)])
+                .from_(expression.args["from"])
+                .where(join_conditions)
+            )
             expression_copy.args["joins"][index].set("side", "right")
+            expression_copy.args["where"] = exp.Where(this=exp.Exists(this=anti_join_clause).not_())
             expression_copy.args.pop("with", None)  # remove CTEs from RIGHT side
+            expression.args.pop("order", None)  # remove order by from LEFT side
 
```

Comment:
Can we test this too in the existing unit test, as is done for `LIMIT`?

Reply:
Do you mean from this:
```
        self.validate_all(
            "SELECT * FROM x LEFT JOIN y ON x.id = y.id UNION ALL SELECT * FROM x RIGHT JOIN y ON x.id = y.id WHERE NOT EXISTS(SELECT 1 FROM x WHERE x.id = y.id) LIMIT 0",
            read={
                "postgres": "SELECT * FROM x FULL JOIN y ON x.id = y.id LIMIT 0",
            },
        )
```
to this:
```
        self.validate_all(
            "SELECT * FROM x LEFT JOIN y ON x.id = y.id UNION ALL SELECT * FROM x RIGHT JOIN y ON x.id = y.id WHERE NOT EXISTS(SELECT 1 FROM x WHERE x.id = y.id) ORDER BY id LIMIT 0",
            read={
                "postgres": "SELECT * FROM x FULL JOIN y ON x.id = y.id ORDER BY id LIMIT 0",
            },
        )
```
?

Reply:
Yeah reusing the existing test should be fine imo, as long as it's a valid Postgres query and shows that the query modifiers (order, limit etc) are preserved once